compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -793,6 +793,9 @@ class JSCodeGen()(using genCtx: Context) {
         name.name
     }.toSet
 
+    val staticNames = moduleClass.companionClass.info.allMembers
+      .collect { case d if d.name.isTermName && d.symbol.isScalaStatic => d.name }.toSet
```

Comment:
That will only take the *simple* names into account, but not the signatures. Is that what the JVM back-end would do? That seems weird, doesn't it?

Reply:
Yeah.  I added the extra static condition, to restricted to what I was fixing, but other original just logs and moves on on any conflict:

https://github.com/lampepfl/dotty/blob/fa93c73909ede9f794f004e0c5e1124a59fbe58e/compiler/src/dotty/tools/backend/jvm/BCodeHelpers.scala#L488-L500

Perhaps it's naive or short-sighted to ignore signatures, but I think it's that way to bias against the generation of forwarders at the first sign of trouble.  What's the API (or where do I find it) to take signatures into account?

Reply:
If the JVM does not take signatures into account, then we shouldn't either on JS.

Otherwise, the best way to actually take signatures into account would be to compare JVM descriptors / IR `MethodName`s.